yjit/src/backend/ir.rs

```diff
@@ -1099,12 +1101,18 @@ impl Assembler
         let gc_offsets = self.compile_with_regs(cb, alloc_regs);
 
         #[cfg(feature = "disasm")]
-        if get_option!(dump_disasm) == DumpDisasm::All || (get_option!(dump_disasm) == DumpDisasm::Inline && cb.inline()) {
+        if let Some(dump_disasm) = get_option_ref!(dump_disasm) {
             use crate::disasm::disasm_addr_range;
             let last_ptr = cb.get_write_ptr();
             let disasm = disasm_addr_range(cb, start_addr, last_ptr.raw_ptr() as usize - start_addr as usize);
             if disasm.len() > 0 {
-                println!("{disasm}");
+                match dump_disasm {
+                    DumpDisasm::Stdout => println!("{disasm}"),
+                    DumpDisasm::File(path) => {
+                        let mut f = File::options().append(true).create(true).open(path).unwrap();
+                        f.write_all(disasm.as_bytes()).unwrap();
+                    }
```

Comment:
Could you move this into a helper function in `disasm.rs` ? I don't really like importing `File` and `Write` into `ir.rs`.

Reply:
sure, done https://github.com/ruby/ruby/pull/6552/commits/40cfe7ed7a3ca10d2cf7ebce5a15d2968e329523